### DISCUSSION TITLE
Querying for a single entry

### DISCUSSION THREAD

**Discussion by PeterPilley** [opensource9ja/danfojs#432] [Q&A]
Hi firstly thanks so much for this library it is exactly what I was after.

I am having some issues getting started though, coming from pandas in python.

I have tried loading json data as dfd.readJSON(file) and then attempting to query it using .query 

df.query({column: ["filename"], is: "==", to:"f216d9f9a9d21f7899f7294854a6f30c.mp4"})

However when I attempt this I am getting the following error:

rows parameter must be an Array. For example: rows: [1,2] or rows: ["0:10"]

I have gone through the documentation and I can see using the rows param for loc and iloc but for query I cant figure out where I am going wrong. any help is really appreciated

**Comment by PeterPilley** [comment]
Been playing with the data a little bit more to see if I can get the query working without success. But I see in the error a .query is performing a .iloc which does have a rows parameter. I have re-read other issues and I cant see where I am going wrong here with this simple query.

I have tried 

df.query({"column": "filename", "is": "==", "to": "f216d9f9a9d21f7899f7294854a6f30c.mp4"})
as well as df.query({rows: ["1:"], column: "filename", is: "==", to:"f216d9f9a9d21f7899f7294854a6f30c.mp4"})

I have tried loading the json via readJSON as well as manually via require('./file.json') and then loading that into a dataframe.

**Reply by risenW** [reply]
Hello @PeterPilley You should be using the new syntax for query. For example instead of `df.query({"column": "filename", "is": "==", "to": "f216d9f9a9d21f7899f7294854a6f30c.mp4"})`, you should do the following:

```javascript
let result = df.query(df["filename"].eq("f216d9f9a9d21f7899f7294854a6f30c.mp4"))

```

See: https://danfo.jsdata.org/api-reference/dataframe/danfo.dataframe.query

**Reply by PeterPilley** [reply]
Awesome thanks so much @risenW, I have the query working now. I really appreciate the quick response

I checked the docs and didnt see a .eq operator, rereading I can see it says using logical operators is for the older version. 

Interestingly if I search for a invalid value, instead of returning an empty result it returns throw new Error("ParamError: end must be greater than start") is this intended?
